core/AGENTS.md

```diff
@@ -0,0 +1,95 @@
+# AGENTS.md
+
+## BigCommerce Catalyst Codebase Overview
+
+This document provides guidance for Large Language Models (LLMs) working with the BigCommerce Catalyst codebase, focusing on the Next.js application architecture, data fetching patterns, and key design principles.
+
+## Repository Structure
+
+The main Next.js application is located in the `/core` directory, which contains the complete e-commerce storefront implementation. Other packages exist outside of `/core` but are not the primary focus for most development work.
+
+## Middleware Architecture
+
+The application uses a composed middleware stack that significantly alters the default Next.js routing behavior. The middleware composition includes authentication, internationalization, analytics, channel handling, and most importantly, custom routing.
+
+### Custom Routing with `with-routes` Middleware
+
+The `with-routes` middleware is the most critical component that overrides Next.js's default path-based routing. Instead of relying on file-based routing, this middleware:
+
+1. **Queries the BigCommerce GraphQL API** to resolve incoming URL paths to specific entity types (products, categories, brands, blog posts, pages).
+
+2. **Rewrites requests** to internal Next.js routes based on the resolved entity type.
+
+3. **Handles redirects** automatically based on BigCommerce's redirect configuration.
+
+This means that URLs like `/my-product-name` can resolve to `/en/product/123` internally, providing flexible URL structure while maintaining SEO-friendly paths.
+
+## Data Fetching and Partial Prerendering (PPR)
+
+### PPR Configuration
+
+The application uses Next.js Partial Prerendering with incremental adoption. This allows static parts of pages to be prerendered while dynamic content streams in.
+
+### Streamable Pattern
+
+A custom `Streamable` utility provides suspense-friendly data fetching that works seamlessly with PPR. This pattern:
+
+- Wraps promises in a lazy-loading container
+- Provides stable promise instances for identical inputs
+- Integrates with React's `use()` hook and Suspense boundaries
+
+### Data Fetching Best Practices
+
+1. **Use React's `cache()` function** for server-side data fetching to ensure deduplication.
```

Comment:
🍹 Same sentence but uses language closer to the official React docs which I think helps the agent

```suggestion
1. **Use React's `cache()` function** for server-side data fetching to memoize function results and prevent repeated fetches or computations **per request** (React will invalidate the cache for all memoized functions for each server request).
```